src/binary.ts

```diff
@@ -219,7 +257,7 @@ export class Binary extends BSONValue {
 
   toUUID(): UUID {
     if (this.sub_type === Binary.SUBTYPE_UUID) {
-      return new UUID(this.buffer.slice(0, this.position));
+      return new UUID(this.buffer.subarray(0, this.position));
```

Comment:
I actually think that we should return a copy here, a new UUID should own its bytes.

Reply:
We can file a ticket about auditing copying/viewing but for now I am fixing that this is a view in node and copy in the browser, changing it to always copy has performance implications and shouldn't be done as a "fix" unless there's a bug caused by it. It would be a feature to "ensure uuid has it's own copy of bytes" if that's what we want from this API.

I can file that, does that sound good?

Reply:
Yup, that sounds good.

Reply:
https://jira.mongodb.org/browse/NODE-6548